src/main/java/se/liu/ida/hefquin/engine/queryplan/logical/LogicalPlanUtils.java

```diff
@@ -55,4 +58,43 @@ public void visit( final LogicalOpUnion op ) {
 		}
 	} // end of class SourceAssignmentChecker
 
+	public static void printStringOfFm( final StringBuilder builder, final FederationMember fm ) {
+
+		final DataRetrievalInterface intFace = fm.getInterface();
+		if ( intFace instanceof SPARQLEndpointInterface){
+			builder.append( ((SPARQLEndpointInterface) intFace).getURL() );
+		}
+		else {
+			builder.append( " Print the Federation Member in the type of interface: " + intFace.getClass().getName() + "is an open TODO" );
+		}
+
+	}
+
+	public static void printTriplesOfGraphPattern(final Op op, final StringBuilder builder ) {
```

Comment:
Did you check whether Jena already has something for printing `Op` objects? If so, I would prefer to simply (re)use that one.